Server/controllers/monitorController.js

```diff
@@ -214,6 +214,10 @@ const editMonitor = async (req, res, next) => {
 
   try {
     const editedMonitor = await req.db.editMonitor(req, res);
+    // Delete the old job(editedMonitor has the same ID as the old monitor)
+    req.jobQueue.deleteJob(editedMonitor);
+    // Add the new job back to the queue
+    req.jobQueue.addJob(editedMonitor._id, editedMonitor);
```

Comment:
Similarly, we could this here like this:

```
// Add the new job back to the queue
    try {
      await req.jobQueue.addJob(editedMonitor._id, editedMonitor);
    } catch (error) {
      // Log the error and continue, as the monitor was still successfully edited
      console.error('Failed to add job to queue:', error);
    }
```

Reply:
If we have a separate try/catch block for each async call in the function then if any one call fails, the others will still execute. 

That would lead to jobs possibly being duplicated or deleted and not added back in.  

```
const doOperation = () => {
  try {
    // delete old job
  } catch (error) {
    console.log(error);
  }
  try {
    // add new job
  } catch (error) {
    console.log(error);
  }
  ```
  
If we do this and deleting the job fails, we will still add the job again, and now we've got duplicate jobs, or we'll throw a different error all togehter.
  
 What we want in this case is one try/catch block with all the risky operations in the try block, so if any of them fails, the whole thing fails

Reply:
Oh, I thought the error in inner try/catch propagates and makes others fail as well and makes finding errors more detailed!
Cool then 👍🏻

Reply:
It does allow us to do more detailed error handling, but control returns to the calling function after the catch block and the code continues to run.

In cases where we do need to do something specific with the error handling we can catch the error, do whatever we need to do with it, and then throw it again.

That will short circuit the function as intended and allow the middleware to handle the error upstream.

I don't think it's necessary here though, the error itself will already contain enough information for debugging I believe.

The client should not receive errors from the JobQueue, these errors are for our debugging use only I believe.